### ISSUE TITLE
Decide how to handle multiple subdomains

### ISSUE THREAD

**Issue by emk** [language-learners/phpbb#20]
We ultimately want to host multiple subdomains on a single server:

- forum.example.com
- www.example.com
- The SuperChallenge bot, etc.

There are two ways we can do this:

- We're using ECS (the Amazon "EC2 Container Service") to manage Docker containers. This integrates nicely with Amazon's ALB ("Application Load Balancer") service, which can direct traffic from various domains to the appropriate container. Unfortunately, this costs roughly $20/month, though it would work for all of our sites.
- We could set up a containerized HTTP proxy directly on the server. This is free, but we would have to make a bunch of decisions. For example, does this mean that we need to link _every_ container we run into a single EC2 "service", with an accompanying `nginx` container? Or can we find a smart proxy that uses Docker tags to direct traffic automatically?

**Comment by emk** [comment]
[An automated nginx reverse proxy](http://jasonwilder.com/blog/2014/03/25/automated-nginx-reverse-proxy-for-docker/) might be what we want. But I wonder whether somebody has written one of these in Go and/or Rust that's all shiny and pre-packaged?

**Comment by emk** [comment]
Also, it would be nice if whatever proxy we use has integrated support for LetsEncrypt, which would also fix #19. Need to do more research. But it might be possible to [do something with this](https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion). Sigh. This would really be nicer as a fully automated service.

**Comment by emk** [comment]
Another option would be to cough up $20/month for an AWS [Application Load Balancer](https://aws.amazon.com/elasticloadbalancing/applicationloadbalancer/), which works quite nicely—but which is also massive overkill for our needs.

**Comment by rdearman** [comment]
Have you looked at this docker container: https://github.com/jwilder/nginx-proxy
It also has a "companion" https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion which will automatically update the encryption with LetsEncrypt.

**Comment by emk** [comment]
Yup. That's what I'm going to set up. Real soon now, I really really really hope. :-/

**Comment by emk** [comment]
OK, I'm theoretically working on this now. Or at least I'm writing this comment, in hopes of motivating myself. That works, right?

And I'll probably get interrupted shortly. Yup, there we go. Well, I'll keep trying.

**Comment by emk** [comment]
OK, here's what I managed to finish so far today: https://gist.github.com/emk/7a03185fa79c04fdca9dbb5d2c221408

This is the `taskdef.json` file we need to use this on ECS, as best as I can assemble it from various documentation. This specifies all the arguments that we would normally pass on the command line.
